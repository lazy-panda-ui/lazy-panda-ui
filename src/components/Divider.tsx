import React from 'react';
import { View, StyleSheet, ViewStyle } from 'react-native';
import { useTheme, Theme } from '../theme/ThemeProvider';

export type DividerVariant = 'solid' | 'dashed' | 'dotted';
export type DividerOrientation = 'horizontal' | 'vertical';
export type DividerThickness = 'thin' | 'regular' | 'thick';

export interface DividerProps {
  /**
   * Visual style variant of the divider
   * @default 'solid'
   */
  variant?: DividerVariant;
  /**
   * Orientation of the divider
   * @default 'horizontal'
   */
  orientation?: DividerOrientation;
  /**
   * Thickness of the divider
   * @default 'regular'
   */
  thickness?: DividerThickness;
  /**
   * Color of the divider, defaults to theme.colors.outline
   */
  color?: string;
  /**
   * Spacing around the divider
   * @default 'md'
   */
  spacing?: keyof Theme['spacing'];
  /**
   * Additional styles for the divider
   */
  style?: ViewStyle;
  /**
   * Test ID for testing
   */
  testID?: string;
}

export const Divider: React.FC<DividerProps> = ({
  variant = 'solid',
  orientation = 'horizontal',
  thickness = 'regular',
  color,
  spacing = 'md',
  style,
  testID,
}) => {
  const theme = useTheme();

  const getThicknessValue = (thickness: DividerThickness): number => {
    switch (thickness) {
      case 'thin':
        return StyleSheet.hairlineWidth;
      case 'thick':
        return 2;
      default:
        return 1;
    }
  };

  const getBorderStyle = (variant: DividerVariant) => {
    switch (variant) {
      case 'dashed':
        return {
          borderStyle: 'dashed' as const,
          borderWidth: getThicknessValue(thickness),
          backgroundColor: undefined,
        };
      case 'dotted':
        return {
          borderStyle: 'dotted' as const,
          borderWidth: getThicknessValue(thickness),
          backgroundColor: undefined,
        };
      default:
        return {
          backgroundColor: color || theme.colors.outline,
          height: orientation === 'horizontal' ? getThicknessValue(thickness) : undefined,
          width: orientation === 'vertical' ? getThicknessValue(thickness) : undefined,
          flexGrow: 1,
        };
    }
  };

  const styles = StyleSheet.create({
    divider: {
      alignSelf: 'stretch',
      ...(orientation === 'horizontal' ? {
        marginVertical: theme.spacing[spacing],
      } : {
        marginHorizontal: theme.spacing[spacing],
      }),
      ...getBorderStyle(variant),
    },
  });

  return (
    <View 
      style={[styles.divider, style]}
      testID={testID}
      accessibilityRole="none"
    />
  );
};
