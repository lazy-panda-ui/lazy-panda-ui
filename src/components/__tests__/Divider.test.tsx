import React from 'react';
import { render } from '@testing-library/react-native';
import { Divider } from '../Divider';

describe('Divider', () => {
  it('renders', () => {
    const { toJSON } = render(<Divider />);
    expect(toJSON()).toMatchSnapshot();
  });
});
