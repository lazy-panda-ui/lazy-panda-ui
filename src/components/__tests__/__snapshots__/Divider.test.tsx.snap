// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#E5E5EA",
        "height": 1,
        "marginVertical": 8,
        "width": "100%",
      },
      undefined,
    ]
  }
/>
`;
